oak-jackrabbit-api/src/main/java/org/apache/jackrabbit/api/JackrabbitNode.java

```diff
@@ -87,4 +89,18 @@ void setMixins(String[] mixinNames)
             return null;
         }
     }
+
+    /**
+     * Same as {@link #getParent()}, but instead of throwing an ItemNotFoundException or AccessDeniedException 
+     * just return {@code null}.
+     * @return the parent node, or {@code null} if there is no parent or the parent node is not accessible.
+     * @throws RepositoryException if an error occurs.
+     */
+    default @Nullable Node getParentOrNull() throws RepositoryException {
+        try {
+            return getParent();
+        } catch (ItemNotFoundException | AccessDeniedException e) {
+            return null;
```

Comment:
i would add some debug log message in case of exception that allows to troubleshoot. wdyt?

Reply:
This is just the default implementation, so I would not do that here.